### PR TITLE
add Stdlib::Filesource

### PR DESCRIPTION
**Pull request by b4ldr**

### CODE REVIEW

**Comment by b4ldr** [comment]
rebased and updated to use new type testing as per #856

**Comment by pegasd** [comment]
As per #856, you don't need the test::* manifest anymore, so it can be safely deleted from PR.

**Comment by b4ldr** [comment]
thanks @pegasd i missed that one, updated now

File: spec/type_aliases/filesource_spec.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    describe 'valid handling' do
+      %w[
+        https://hello.com
+        https://notcreative.org
+        https://canstillaccepthttps.co.uk
+        http://anhttp.com
+        http://runningoutofideas.gov
+        file:///hello/bla
+        file:///foo/bar.log
+        puppet:///modules/foo/bar.log
```
Comment (binford2k) [inline spec/type_aliases/filesource_spec.rb]: You should also test `puppet://someserver/modules/foo/bar.log`

Comment (b4ldr) [reply spec/type_aliases/filesource_spec.rb]: @binford2k thanks for the pointer i didn't see that varient in the docs.  Have updeated the type pattern and spec tests

**Comment by david22swan** [comment]
@binford2k Are you happy that the changes you requested have been made?

File: spec/type_aliases/filesource_spec.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          [nil],
+          [nil, nil],
+          { 'foo' => 'bar' },
+          {},
+          '',
+          '*/Users//nope',
+          '\\Users/hc/wksp/stdlib',
+          'C:noslashes',
+          '\\var\\tmp',
+          'puppet:///foo/bar.log',
```
Comment (alexjfisher) [inline spec/type_aliases/filesource_spec.rb]: @binford2k This is actually valid though?  https://puppet.com/docs/puppet/5.5/file_serving.html#whats-a-mount-point-in-a-puppet-uri
We're using this type in puppet-zabbix and a user has complained the regex is rejecting valid puppet fileserver mount points. https://github.com/voxpupuli/puppet-zabbix/issues/580

Comment (binford2k) [reply spec/type_aliases/filesource_spec.rb]: Yes, that should be valid.

**Pull request merged**